scripts/api_wrapper.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+A wrapper script to interface with our manager app via API calls.
+The manager calls this script and passes the request as cmd line arguments.
+This wrapper starts the script described in the command line argument,
+and when that finishes, it makes an API call back to the manager.
+"""
+
+import logging
+import sys
+import requests
+import subprocess
+
+
+def main():
+    step_id = sys.argv[2]
+
+    logging.basicConfig(
+        filename= sys.argv[1],
+        filemode='a',
+        format='%(asctime)s - %(threadName)-10s)- %(levelname)s - %(message)s',
+        level=logging.INFO
+    )
+
+    logging.info(f"Script {sys.argv[3]} (id: {step_id}) - Starting with: {sys.argv[2:]}")
+    results = subprocess.run(sys.argv[3:])
+    if results.returncode == 0:
+        logging.info(f"Script {sys.argv[3]} (id: {step_id}) - Successfully executed")
+        # TODO this url shoud not be hardwired:
+        success_url = f"http://127.0.0.1:8000/completed/{step_id}"

```

Comment:
Please make this URL not hardwired :smile:

Reply:
I spent hours trying to fix this.

The problem is that to dynamically get the current IP address of the webapp itself you have to query that from the fastapi object. 
That is currently created in our main.py:  
`app = FastAPI(title="CC Corpus manager")`

Trying to give access to that object or any variable created in the main.py leads to circular imports error. I tried to work around that by moving it to a third python file, but I still could not avoid the circular imports error.

Just now a new idea came to my mind: maybe it should write its own ip address into the DB on startup, and I could fetch the value from there, and thus avoid the circular imports error...